test/suites/sharing/ram_share_test.go

```diff
@@ -0,0 +1,433 @@
+		Expect(err).To(BeNil())
+
+		// Wait for resource share invitation to appear in primary account
+		var invitation *ram.ResourceShareInvitation = nil
+		Eventually(func(g Gomega) {
+			listInvitationsInput := &ram.GetResourceShareInvitationsInput{}
+			listInvitationsResult, err := primaryRamClient.GetResourceShareInvitations(listInvitationsInput)
+			Expect(err).NotTo(HaveOccurred())
+
+			for _, inv := range listInvitationsResult.ResourceShareInvitations {
+				if *inv.ResourceShareName == randomName && *inv.SenderAccountId == secondaryAccountId {
+					invitation = inv
+					break
+				}
+			}
+			g.Expect(invitation).ToNot(BeNil())
+		}).Should(Succeed())
+
+		// Accept resource share invitation
+		acceptInput := &ram.AcceptResourceShareInvitationInput{
+			ResourceShareInvitationArn: invitation.ResourceShareInvitationArn,
+		}
+		_, err = primaryRamClient.AcceptResourceShareInvitation(acceptInput)
+		Expect(err).ToNot(HaveOccurred())
+
+		// Wait for service network to appear in primary account
+		Eventually(func(g Gomega) {
+			getSNInput := &vpclattice.GetServiceNetworkInput{
+				ServiceNetworkIdentifier: createSNResult.Id,
+			}
+			_, err = primaryVpcLatticeClient.GetServiceNetworkWithContext(ctx, getSNInput)
+			g.Expect(err).ToNot(HaveOccurred())
+		}).WithTimeout(5 * time.Minute).WithPolling(5 * time.Second).Should(Succeed())
+
+		// Create gateway with same name as service network
+		gateway := &gwv1.Gateway{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      randomName,
+				Namespace: k8snamespace,
+			},
+			Spec: gwv1.GatewaySpec{
+				GatewayClassName: "amazon-vpc-lattice",
+				Listeners: []gwv1.Listener{
+					{
+						Name: gwv1.SectionName(randomName),
+						Port:     gwv1.PortNumber(80),
+						Protocol: gwv1.HTTPProtocolType,
+					},
+				},
+			},
+		}
+		err = testFramework.Create(ctx, gateway)
+		Expect(err).ToNot(HaveOccurred())
+
+		Eventually(func(g Gomega) {
+			// Policy status should be Accepted
+			gwNamespacedName := types.NamespacedName{
+				Name:      gateway.Name,
+				Namespace: gateway.Namespace,
+			}
+			gw := &gwv1.Gateway{}
+			err := testFramework.Client.Get(ctx, gwNamespacedName, gw)
+			g.Expect(err).To(BeNil())
+			g.Expect(len(gw.Status.Conditions)).To(BeEquivalentTo(2))
+			g.Expect(gw.Status.Conditions[0].Type).To(BeEquivalentTo(string(gwv1.GatewayConditionAccepted)))
+			g.Expect(gw.Status.Conditions[0].Status).To(BeEquivalentTo(metav1.ConditionTrue))
+			g.Expect(gw.Status.Conditions[0].ObservedGeneration).To(BeEquivalentTo(1))
+			g.Expect(gw.Status.Conditions[0].Reason).To(BeEquivalentTo(string(gwv1.GatewayReasonAccepted)))
+			g.Expect(gw.Status.Conditions[1].Type).To(BeEquivalentTo(string(gwv1.GatewayConditionProgrammed)))
+			g.Expect(gw.Status.Conditions[1].Status).To(BeEquivalentTo(metav1.ConditionTrue))
+			g.Expect(gw.Status.Conditions[1].ObservedGeneration).To(BeEquivalentTo(1))
+			g.Expect(gw.Status.Conditions[1].Reason).To(BeEquivalentTo(string(gwv1.GatewayReasonProgrammed)))
```

Comment:
Most of these are not related to ram share, checking for programmed status is enough.

Reply:
Will remove unrelated ones